DNN Platform/DotNetNuke.Abstractions/Logging/IEventLogger.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Abstractions.Logging
+{
+    using DotNetNuke.Abstractions.Portals;
+
+    /// <summary>
+    /// The event logger provides APIs for adding
+    /// new event logs.
+    /// </summary>
+    public interface IEventLogger
+    {
+        /// <summary>
+        /// Add setting log.
+        /// </summary>
+        /// <param name="logTypeKey">The log type key.</param>
+        /// <param name="idFieldName">The id field name.</param>
+        /// <param name="idValue">The id value.</param>
+        /// <param name="settingName">The setting name.</param>
+        /// <param name="settingValue">The setting value.</param>
+        /// <param name="userId">The user id.</param>
+        void AddSettingLog(EventLogType logTypeKey, string idFieldName, int idValue, string settingName, string settingValue, int userId);
+
```

Comment:
I know @mitchelsellers' already expressed his displeasure before, but this seems like a scenario that'd be perfect for an extension method.  A specific use case that's really just sugar over calling another overload.

Reply:
This one kind of confused me and I wasn't sure where to put it. I'll take a closer look after I look at the rest of the feedback

Reply:
I tried moving this to an extension method but it opens up a lot of circular dependency issues. We have moved interfaces to the Abstractions project and we can't really move the extension method to the abstraction project.

Should I add a new extension method in `DotNetNuke.Library`. If you have thoughts on this feel free to push right to the repository

Reply:
@bdukes @mitchelsellers is there any action to take on this one?

Reply:
I did move this to an extension method.  I created `EventLogExtensions` in `DotNetNuke.Services.Log.EventLog`.